src/server/conn_context.h

```diff
@@ -17,20 +18,24 @@ class EngineShardSet;
 class ConnectionContext;
 class ChannelStore;
 
-struct StoredCmd {
-  const CommandId* descr;
+// Stores command id and arguments for delayed invocation.
+// Used for storing MULTI/EXEC commands.
+class StoredCmd {
+ public:
+  StoredCmd(const CommandId* cid, CmdArgList args);
 
- private:
-  std::unique_ptr<char[]> backing_storage_;
-  CmdArgVec arg_vec_;
-  CmdArgList arg_list_;
+  size_t NumArgs() const;
 
- public:
-  StoredCmd(const CommandId* d, CmdArgList args);
+  // Fill the arg list with stored arguments, it should be at least of size NumArgs().
+  // Between filling and invocation, cmd should NOT be moved.
+  void Fill(CmdArgList args);
```

Comment:
const

Reply:
CmdArgLists consists of `MutableSlice`, that must be constructed from char*, which can be retreived only from a mutable string (buffer_ member)

Reply:
I meant the function is const :)

Reply:
got ya